server/src/main/java/org/elasticsearch/index/IndexFormatVersion.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index;
+
+import org.elasticsearch.Version;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.NavigableMap;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * The index format version.
+ * <p>
+ * Prior to 8.7.0, the node {@link Version} was used everywhere. This class separates the index format version
+ * from the running node version.
+ */
+public final class IndexFormatVersion implements Comparable<IndexFormatVersion> {
```

Comment:
This is quite basic at the moment; we can add functionality as we expand uses of this class